src/commands/skip.rs

```diff
@@ -20,9 +23,43 @@ pub async fn skip(
 
     if queue.is_empty() {
         return create_response(&ctx.http, interaction, NOTHING_IS_PLAYING).await;
-    } else if queue.skip().is_ok() {
-        return create_response(&ctx.http, interaction, SKIPPED).await;
     }
 
+    let mut data = ctx.data.write().await;
+    let cache_map = data.get_mut::<GuildCacheMap>().unwrap();
+
+    let cache = cache_map.entry(guild_id).or_default();
+
+    if !cache.current_skip_votes.contains(&interaction.user.id) {
+        cache.current_skip_votes.push(interaction.user.id);
+    }
+
+    let guild = ctx.cache.guild(guild_id).await.unwrap();
+    let skip_threshold = guild.voice_states.len() / 2;
+
+    let message = if cache.current_skip_votes.len() >= skip_threshold && queue.skip().is_ok() {
+        SKIPPED.to_string()
+    } else {
+        format!(
+            "{}{} {} {} {}",
+            SKIP_VOTE_EMOJI,
+            interaction.user.id.mention(),
+            SKIP_VOTE_USER,
+            skip_threshold - cache.current_skip_votes.len(),
+            SKIP_VOTE_MISSING
+        )
+    };
+
+    create_response(&ctx.http, interaction, &message).await?;
+    drop(data);
+
     Ok(())
 }
+
+pub async fn forget_skip_votes(data: &Arc<RwLock<TypeMap>>, guild_id: GuildId) {
+    let mut data = data.write().await;
+    let cache_map = data.get_mut::<GuildCacheMap>().unwrap();
+
+    let cache = cache_map.get_mut(&guild_id).unwrap();
+    cache.current_skip_votes = Vec::new();
+}
```

Comment:
This should probably return a Result/Option so we can cut some unwraps.

Reply:
Why would we want to return an option here exactly?

Reply:
Because fetching from the context might fail so instead of unwrapping we propagate the error upwards and leave it to the caller to handle errors (or unwrap as well).